common/src/main/java/org/apache/sedona/common/utils/RasterUtils.java

```diff
@@ -80,6 +79,89 @@ public static GridCoverage2D create(WritableRaster raster, GridGeometry2D gridGe
         return create(raster, gridGeometry, bands, null);
     }
 
+    /**
+     * Create a new raster, cloning all metadata from the passed reference raster.
+     * @param raster The raster object to be wrapped as an image.
+     * @param referenceRaster The reference raster to clone, all metadata is cloned
+     * @param noDataValue Optionally, a new noDataValue, if passed null, noDataVal of the reference raster is used.
+     * @return cloned raster
+     */
+    public static GridCoverage2D create(WritableRaster raster, GridCoverage2D referenceRaster, Double noDataValue, boolean keepMetadata) {
+        GridSampleDimension[] bands = keepMetadata ? referenceRaster.getSampleDimensions() : null;
+        if (Objects.isNull(noDataValue)) {
+            noDataValue = RasterBandAccessors.getBandNoDataValue(referenceRaster, 1); //resort to using noDataValue of the original raster.
+        }
+        Map propertyMap = null;
+        if (keepMetadata) {
+            propertyMap = referenceRaster.getProperties();
+        }
+        ColorModel originalColorModel = referenceRaster.getRenderedImage().getColorModel();
+        GridGeometry2D gridGeometry = referenceRaster.getGridGeometry();
+        int numBand = raster.getNumBands();
+        int rasterDataType = raster.getDataBuffer().getDataType();
+
+        final ColorSpace cs = new BogusColorSpace(numBand);
+        final int[] nBits = new int[numBand];
+        Arrays.fill(nBits, DataBuffer.getDataTypeSize(rasterDataType));
```

Comment:
I believe that these could also be moved to the else branch

Reply:
Sorry about that! 
Pushed the change